packages/www/components/Dashboard/WebhookDialog/index.tsx

```diff
@@ -54,6 +54,11 @@ const eventOptions = [
   "multistream.connected",
   "multistream.error",
   "multistream.disconnected",
+  "asset.created",
```

Comment:
They are enums but as a string literal union instead, as I explained in the other comment.

Unfortunately, the schema definition only becomes a Typescript type, which has no runtime value in itself that we could use here instead of redeclaring the possible events. This is also useful because we don't want every event to be available in the frontend, just like we comment out some of the values here like `stream.detection` and `playback.user.new` which are kept as a little more internal features for now.

Also, enums wouldn't solve this completely, since we'd still need some hacks to actually get a list of values as we need here: https://stackoverflow.com/a/48768775

If we were actually writing all the code from our schema objets, we could declare first the array and then derive the string literals from it ([like this](https://dev.to/remshams/derive-union-of-string-literal-types-with-lookup-types-in-typescript-1kkf)), but we don't. Our typescript schema object definitions are generated dynamically [using ajv](https://github.com/livepeer/livepeer-com/blob/7eeb04d5bdd819a5d7022dd6455a2b51d15fbbfb/packages/api/src/compile-schemas.js#L53), so I don't think it's worth moving away from that.

Reply:
Oh, regarding this being used on the queue publish requests, this code is in the frontend (`www`) package, while the queue publish happens in the `api`. We normally try to keep the cross-package dependencies to a minimum, so currently there are only "type" dependencies and then [1 specific file](https://github.com/livepeer/livepeer-com/blob/a82d714ca7b801d9b6c8dc83fbb52996fd2b0b36/packages/api/src/hash.js#L1) for hashing passwords, which is useful to have a consistent code between front and backend. 

It could be done by declaring these on the `api` and then depending on it here (just so we don't have a cyclic dependency). But then that would also mean be a new code dependency between packages just to avoid this repetition (which would still need a filter anyway for the "private" webhooks), additionally from having to move away from schema types (and validators) code generation, which saves us a lot of manual fiddling of types as well.

So overall/tldr, I don't think it's worth moving away from string literals to some sort of manual enum here. But I've improved this list by at least adding a type to it, to make sure that every value on the array will actually match an existing webhook event. It won't check if every value is here nor if we don't have duplicates, but I think that wouldn't be guaranteed on any "enum" implementation anyway.